exercises/say/.meta/src/reference/java/Say.java

```diff
@@ -0,0 +1,204 @@
+import java.util.Arrays;
+
+public class Say {
+
+    public String say(long number) {
+        if (number < 0) {
+            throw new IllegalArgumentException("input out of range");
+        }
+        char[] digits = String.valueOf(number).toCharArray();
```

Comment:
Rather than processing a list of characters, why not turn the input number into a long and then we can extract the numbers using numeric operations (eg. integer division and modulo)? This also gets rid of the leading zeros problem that you had to address multiple times in this solution.